### PR TITLE
Do not pass oauth2 as a username to Git credentials for Bitbucket

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Set `bitbucket-*****` as a token name annotation for bitbucket token secret. This is needed to pass username instead of `oauth2` for bitbucket credentials:
https://github.com/eclipse-che/che-server/blob/7dc7a6151144b1da05ed8d6991f985db055d4440/infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesGitCredentialManager.java#L169-L184

This fixes the bug when a private bitbucket.org project is not cloned.

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-5952

### How to test this PR?

1. Deploy che with the PR image: `quay.io/eclipse/che-server:pr-662
2. [Setup bitbucket.org oauth](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-the-bitbucket-cloud/)
3. Start a workspace from a private bitbucket.org repository.
See: the workspace starts successfully.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java [hunk 1]
```diff
@@ -111,7 +111,9 @@ public Response callback(UriInfo uriInfo, @Nullable List<String> errorValues)
               EnvironmentContext.getCurrent().getSubject().getUserId(),
               null,
               null,
-              NameGenerator.generate(OAUTH_2_PREFIX, 5),
+              NameGenerator.generate(
+                  // Do not pass oauth2 as a username to Git credentials for Bitbucket
+                  "bitbucket".equals(providerName) ? providerName + "-" : OAUTH_2_PREFIX, 5),
```
Comment (ibuziuk) [inline wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:116]: please, extract the logic to the separate method with a proper javadoc

Comment (ibuziuk) [reply wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:116]: @vinokurig is the providee name the same for on-prem Bitbucket server and SaaS ?

Comment (vinokurig) [reply wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java]: yes, it is the same. We have `bitbucket-server` implementation but it handles oauth1 only and does not use this logic. OAuth2 flow from both bitbucket and bitbucket server is handled by the `bitbucket` implementation.

**Review by ibuziuk — APPROVED** [review]

**Comment by artaleks9** [comment]
Verified on Eclipse Che with `quay.io/eclipse/che-server:pr-662` - the functionality works as expected.

**Review by artaleks9 — APPROVED** [review]

**Pull request merged**